app/src/main/report/components/Reports/ReportPageTitle.tsx

```diff
@@ -0,0 +1,34 @@
+import {reportStore} from "../../stores/ReportStore";
+import {connectToStores} from "../../../shared/alt";
+import * as React from "react";
+
+interface Props {
+    name: string;
+    displayName: string;
+}
+
+export class ReportPageTitleComponent extends React.Component<Props, undefined> {
+    static getStores() {
+        return [reportStore];
+    }
+
+    static getPropsFromStores(): Props {
+        const s = reportStore.getState();
+        const props: Props = {
+            name: s.currentReport,
+            displayName: null
+        };
+        const details = s.versionDetails[s.currentReport];
+        if (details) {
```

Comment:
would it ever be the case that the details have not  been fetched yet, and if so should we wait for them to be fetched before displaying?

Reply:
Yes, it could happen: When you navigate to the page is when we kick off the API request to fetch the details. So we already have a loading spinner for the page content. I considered just making the title "Loading..." until the details have been fetched, but I figure if it does for some reason take a few seconds (due to slow Internet) it would be more informative for the user if we displayed the information we already had, and then switch to the better title when it arrived.

It most usage I think it will be very quick and the user won't really see the shorter title.